lib/galaxy/webapps/galaxy/api/datatypes.py

```diff
@@ -2,14 +2,200 @@
+            return self._mapping()
+        except Exception as e:
+            log.exception('Could not get data type mapping')
+            raise e
+
+    @router.get('/api/datatypes/types_and_mapping',
+        summary="Returns all the data types extensions and their mappings",
+        response_model=DatatypesCombinedMap,
+        response_description="Dictionary to map data types with their classes")
+    async def types_and_mapping(self) -> DatatypesCombinedMap:
+        """Combines the datatype information from (/api/datatypes) and the
+        mapping information from (/api/datatypes/mapping) into a single
+        response."""
+        return DatatypesCombinedMap(datatypes=self._index(), datatypes_mapping=self._mapping())
+
+    @router.get('/api/datatypes/sniffers',
+        summary="Returns the list of all installed sniffers",
+        response_model=List[str],
+        response_description="List of datatype sniffers")
+    async def sniffers(self) -> List[str]:
+        """Gets the list of all installed data type sniffers."""
+        try:
+            return self._sniffers()
+        except Exception as e:
+            log.exception('Could not get sniffers')
+            raise e
+
+    @router.get('/api/datatypes/converters',
+        summary="Returns the list of all installed converters",
+        response_model=List[DatatypeConverter],
+        response_description="List of all datatype converters")
+    async def converters(self) -> List[DatatypeConverter]:
+        """Gets the list of all installed converters."""
+        try:
+            return self._converters()
+        except Exception as e:
+            log.exception('Could not get converters')
+            raise e
+
+    @router.get('/api/datatypes/edam_formats',
+        summary="Returns a dictionary/map of datatypes and EDAM formats",
+        response_model=Dict[str, str],
+        response_description="Dictionary/map of datatypes and EDAM formats")
+    async def edam_formats(self) -> Dict[str, str]:
+        """Gets a map of datatypes and their corresponding EDAM formats."""
+        try:
+            return self.datatypes_registry.edam_formats
+        except Exception as e:
+            log.exception('Could not get EDAM format mappings')
+            raise e
+
+    @router.get('/api/datatypes/edam_data',
+        summary="Returns a dictionary/map of datatypes and EDAM data",
+        response_model=Dict[str, str],
+        response_description="Dictionary/map of datatypes and EDAM data")
+    async def edam_data(self) -> Dict[str, str]:
+        """Gets a map of datatypes and their corresponding EDAM data."""
+        try:
+            return self.datatypes_registry.edam_data
+        except Exception as e:
+            log.exception('Could not get EDAM data mappings')
+            raise e
+
+    def _index(self, extension_only: Optional[bool] = True, upload_only: Optional[bool] = True) -> Union[List[DatatypeDetails], List[str]]:
+        if extension_only:
```

Comment:
I would appreciate it if you can convert these to like view functions that can be reuse in the existing controller below.

I know we plan to eliminate those controllers but I don't think we should duplicate hundreds of lines of code even in the short term in case those plans don't pan out. I added a bunch of stuff to https://github.com/galaxyproject/galaxy/pull/10823 so that those controllers can return pydantic models and exceptions handling is synchronized.

If the old code and the new code are both as thin as possible - those are ideal controllers IMO and they will allow us to duplicate as little as possible.

Reply:
Of course! that was one of my concerns in the PR comment. Thanks for the suggestions, I will do it :)

Reply:
Done :)